pkg/controller/launch/pool_manager.go

```diff
@@ -113,13 +135,31 @@ func (cpm *ConcretePoolManager) SetStatus(status *PoolStatus) error {
 		return err
 	}
 
-	for i, curInfo := range copy.Status.NodePools {
-		if curInfo.Name == newInfo.Name {
-			if curInfo == newInfo {
+	// Add new pools in the spec to the status
+	for _, specPool := range copy.Spec.NodePools {
+		// Find the pool
+		if npi, ok := nodePoolInfoGet(copy.Status.NodePools, specPool.Name); ok {
+			// is there a need to update?
+			if copy.Status.NodePools[npi] == newInfo {
 				return nil
 			}
+			copy.Status.NodePools[npi] = newInfo
+			_, err = cpm.Clients.Kubernikus.Kubernikus().Klusters(copy.Namespace).Update(copy)
+			return err
+			// not found so add it
+		} else {
+			copy.Status.NodePools = append(copy.Status.NodePools, newInfo)
+			_, err = cpm.Clients.Kubernikus.Kubernikus().Klusters(copy.Namespace).Update(copy)
+			return err
+		}
+	}
 
-			copy.Status.NodePools[i] = newInfo
+	// Delete pools from the status that are not in spec
+	for i, infoPool := range copy.Status.NodePools {
+		// find the pool
+		if _, ok := nodePoolSpecGet(copy.Spec.NodePools, infoPool.Name); !ok {
+			// not found in the spec anymore so delete it
+			copy.Status.NodePools = removeNodePool(copy.Status.NodePools, i)
 			_, err = cpm.Clients.Kubernikus.Kubernikus().Klusters(copy.Namespace).Update(copy)
```

Comment:
We have a helper for updating a kluster. It would require a bit of refactoring though. Maybe as a second step... https://github.com/sapcc/kubernikus/blob/master/pkg/util/kluster.go#L16-L18